synapse/handlers/message.py

```diff
@@ -336,6 +337,25 @@ def create_event(self, requester, event_dict, token_id=None, txn_id=None,
             prev_events_and_hashes=prev_events_and_hashes,
         )
 
+        # In an ideal world we wouldn't need the second part of this condition. However,
+        # this behaviour isn't spec'd yet, meaning we should be able to deactivate this
+        # behaviour, and this code is evaluated each time a new m.room.aliases event is
+        # created, which includes hitting a /directory route, therefore not including it
```

Comment:
I don't really understand this "therefore not including it" part. What is "it": the condition, or the check code?

Long sentence is long.